lib/bloc/user_profile/profile_animal.dart

```diff
@@ -0,0 +1,238 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations_en.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations_es.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations_fi.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations_pt.dart';
+
+enum ProfileAnimal {
+  BAT,
+  BEAR,
+  BOAR,
+  CAT,
+  CHICK,
+  COW,
+  DEER,
+  DOG,
+  EAGLE,
+  ELEPHANT,
+  FOX,
+  FROG,
+  HIPPO,
+  HUMMINGBIRD,
+  KOALA,
+  LION,
+  MONKEY,
+  MOUSE,
+  OWL,
+  OX,
+  PANDA,
+  PIG,
+  RABBIT,
+  SEAGULL,
+  SHEEP,
+  SNAKE,
+}
+
+ProfileAnimal profileAnimalFromName(String name, AppLocalizations texts) {
```

Comment:
Isn't is better to build the global map (with all locales) once as a static map and when an animal is searched just use it?
The advantage is that it will faster for searching and smaller code.

Reply:
True, I did apply it :)